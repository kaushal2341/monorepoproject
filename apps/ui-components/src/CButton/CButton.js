import React, { memo } from 'react'
import { Button } from 'react-bootstrap'
import PropTypes from 'prop-types'
const CButton = props => {
    const {
        id,
        name,
        children,
        variant,
        active,
        classes,
        disabled,
        as,
        onClickHandler,
        href,
        size,
        type,
        block,
        bsPrefix
    } = props;
    return (
        <>
            <Button
                id={id}
                children={children}
                variant={variant}
                active={active}
                className={classes}
                disabled={disabled}
                as={as}
                onClick={onClickHandler}
                href={href}
                size={size}
                type={type}
                block={block}
                bsPrefix={bsPrefix}
            >
                {name}
            </Button>
        </>
    )
};


CButton.defaultProps = {
    name: 'Save',
    type: 'button',
    onClickHandler: () => {}
};

CButton.propTypes = {
    id: PropTypes.string.isRequired || PropTypes.number.isRequired,
    children: PropTypes.element,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    onClickHandler: PropTypes.func.isRequired,
    variant: PropTypes.string,
    active: PropTypes.bool,
    classes: PropTypes.string,
    disabled: PropTypes.bool,
    href: PropTypes.string,
    bsPrefix: PropTypes.string,
    size: PropTypes.string,
    block: PropTypes.bool
};

export default memo(CButton);