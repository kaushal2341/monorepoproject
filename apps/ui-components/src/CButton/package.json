{
    "name": "@cogent/c-button",
    "version": "0.0.0",
    "main": "CButton.js",
    "license": "MIT",
    "private": true
}
