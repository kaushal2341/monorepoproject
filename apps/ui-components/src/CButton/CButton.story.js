import React from 'react';
import {storiesOf} from "@storybook/react";

import CButton from './CButton';

storiesOf('CButton', module)
    .add('default CButton', () => <CButton>Bijay</CButton>);