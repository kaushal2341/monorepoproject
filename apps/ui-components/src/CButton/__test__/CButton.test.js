import CButton from '../CButton';

expect.addSnapshotSerializer(enzymeSerializer);
describe('CButton Component', () => {
    let wrapper, wrapper1, mockFunction;
    beforeEach(() => {
        mockFunction = jest.fn();
        wrapper = shallow(<CButton id="c-button" onClickHandler={mockFunction}/>)
        wrapper1 = mount(<CButton id="c-button"/>);

    });
    afterEach(() => {
        wrapper1.unmount()
    })
    it('should be defined', () => {
        expect(wrapper).toBeDefined();
    });

    it('should have only one button', () => {
        expect(wrapper.find('#c-button')).toHaveLength(1);
    });

    it('should have show name of button when props name is given', () => {
        wrapper.setProps({
            name: 'Submit'
        });
        expect(wrapper.find('#c-button').text()).toBe('Submit')
    })

    it('should show default name of button when props name is not given', () => {
        expect(wrapper.find('#c-button').text()).toBe('Save');
    });

    it('should have type button when props type is not given', () => {
        expect(wrapper.find('#c-button').prop('type')).toBe('button');
    });

    it('should have give props available to the button', () => {
        const buttonProps =
            ['id', 'type',
                'active', 'className',
                'children', 'as',
                'block', 'disabled',
                'href', 'size',
                'variant', 'bsPrefix',
                'onClick'];
        const actualButtonProps = Object.keys((wrapper.find('#c-button').props()));
        buttonProps.map((bprop, index) => {
            expect(buttonProps).toContain(actualButtonProps[index]);
        });
    });

    it('should have call the function when the button is clicked', () => {
        wrapper.find('#c-button').simulate('click');
        expect(mockFunction).toHaveBeenCalled();
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(wrapper1).toMatchSnapshot();
        })
    });
});