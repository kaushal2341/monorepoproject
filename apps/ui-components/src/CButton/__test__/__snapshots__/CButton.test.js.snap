// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CButton Component Snapshot Testing should match snapshot 1`] = `
<CButton
  id="c-button"
>
  <Button
    active={false}
    disabled={false}
    id="c-button"
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      id="c-button"
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </Button>
</CButton>
`;
